src/main/java/reposense/configparser/InputParameter.java

```diff
@@ -0,0 +1,58 @@
+package reposense.configparser;
+
+import java.io.File;
+import java.util.Date;
+import java.util.Optional;
+
+/**
+ * Stores parsed user-supplied CLI arguments.
+ */
+public final class InputParameter {
+    private static final String DEFAULT_FILE_ARG = ".";
+
+    private File configFile;
+
+    private File targetFile;
+
+    private Optional<Date> sinceDate;
```

Comment:
I don't think we should be using `Optional` as a field, but only as a method return type, as discussed in this [SO post](https://stackoverflow.com/questions/29033518/is-it-a-good-practice-to-use-optional-as-an-attribute-in-a-class).

We just can leave this as type `Date`, since `GitLogger#gitLog(String, Date, Date)` already checks for the `null`.

Reply:
As discussed, we will stick to Optional<Date> for now.